cmd/lambda/accounts/create_test.go

```diff
@@ -1,686 +1,95 @@
-			res,
-		)
-	})
-
-	t.Run("should create a principal role and policy", func(t *testing.T) {
-
-		// Mock the TokenService (assumes role into the user account)
-		tokenServiceMock := &commonMocks.TokenService{}
-		TokenSvc = tokenServiceMock
-
-		// Mock Token Service, to assume adminRoleArn
-		mockAdminRoleSession := &awsMocks.AwsSession{}
-		mockAdminRoleSession.On("ClientConfig", mock.Anything).Return(client.Config{
-			Config: &aws.Config{},
-		})
-		tokenServiceMock.On("NewSession", mock.Anything, "arn:mock").
-			Return(mockAdminRoleSession, nil)
-		tokenServiceMock.On("AssumeRole", mock.Anything).Return(nil, nil)
-
-		// Mock the RoleManager (creates the IAM Role)
-		roleManager := roleManagerMocks.RoleManager{}
-		RoleManager = &roleManager
-
-		// RoleManager should use an IAM client,with the assumed role session
-		roleManager.On("SetIAMClient", mock.Anything)
-
-		// Setup expected AssumeRolePolicy
-		expectedTrustAccount := "0987654321"
-		expectedAssumeRolePolicy := strings.TrimSpace(`
+	type response struct {
+		StatusCode int
+		Body       string
+	}
+	tests := []struct {
+		name       string
+		expResp    response
+		reqBody    string
+		retAccount *account.Account
+		retErr     error
+	}{
 		{
-			"Version": "2012-10-17",
-			"Statement": [
-				{
-					"Effect": "Allow",
-					"Principal": {
-						"AWS": "arn:aws:iam::` + expectedTrustAccount + `:root"
-					},
-					"Action": "sts:AssumeRole",
-					"Condition": {}
-				}
-			]
-		}
-		`)
-
-		// Mock the RoleManager, to create an IAM Role for the Principal user
-		roleManager.On("CreateRoleWithPolicy",
-			mock.MatchedBy(func(input *rolemanager.CreateRoleWithPolicyInput) bool {
-				// Verify the expected input
-				assert.Equal(t, "DCEPrincipal", input.RoleName)
-				assert.Equal(t, "Role to be assumed by principal users of DCE", input.RoleDescription)
-				assert.Equal(t, expectedAssumeRolePolicy, input.AssumeRolePolicyDocument)
-				assert.Equal(t, int64(100), input.MaxSessionDuration)
-				assert.Equal(t, "DCEPrincipalDefaultPolicy", input.PolicyName)
-				assert.Equal(t, []*iam.Tag{
-					{Key: aws.String("Terraform"), Value: aws.String("False")},
-					{Key: aws.String("Source"), Value: aws.String("github.com/Optum/dce//cmd/lambda/accounts")},
-					{Key: aws.String("Environment"), Value: aws.String("DefaultTagEnvironment")},
-					{Key: aws.String("Contact"), Value: aws.String("DefaultTagContact")},
-					{Key: aws.String("AppName"), Value: aws.String("DefaultTagAppName")},
-					{Key: aws.String("Name"), Value: aws.String("DCEPrincipal")},
-				}, input.Tags)
-				assert.Equal(t, true, input.IgnoreAlreadyExistsErrors)
-				assert.Equal(t, "", "")
-
-				return true
-			}),
-		).Return(&rolemanager.CreateRoleWithPolicyOutput{}, nil)
-
-		// Call the controller with the account
-		_, err := Handler(
-			context.TODO(),
-			createAccountAPIRequest(t, CreateRequest{
-				ID:           "1234567890",
-				AdminRoleArn: "arn:mock",
+			name: "success",
```

Comment:
Can you please update the description to Gherkins style? I updated in lease to "When lease service - Get returns success".